parser.cs

```diff
@@ -1469,13 +1466,22 @@ public class Parser {
             case TOKENS.IDENTIFIER:
                 String identifier = __lookahead.Lexeme;
                 KINDS identifierKind = __symbolTableStack.Peek().GetKind(identifier);
-                if(identifierKind == KINDS.VAR || identifierKind == KINDS.PARAMETER) {
+                if(identifierKind == KINDS.VAR) {
                     Console.Write(116 + " ");
                     String factorId = variableIdentifier();
                     TYPES factorType = __symbolTableStack.Peek().GetType(factorId);
                     factorRec = new SemRecord(factorType, factorId);
                     __analyzer.genPushVar(factorRec);
                     return factorRec;
+                } else if (identifierKind == KINDS.PARAMETER) {
```

Comment:
Here I started work, but it needs completion. Basically, what we need to know is what the mode is of the particular entry so we know if we need to push say, 1(D0) or @1(D0)